sklearn/neighbors/tests/test_lmnn.py

```diff
@@ -0,0 +1,677 @@
+                         '`n_neighbors` (=2) is not less than the number of '
+                         'samples in the smallest non-singleton class (=2). '
+                         '`n_neighbors_` will be set to 1 for estimation.',
+                         lmnn.fit, X, y)
+
+
+def test_n_components():
+    X = np.arange(12).reshape(4, 3)
+    y = [1, 1, 2, 2]
+
+    init = np.random.rand(X.shape[1] - 1, 3)
+
+    # n_components = X.shape[1] != transformation.shape[0]
+    n_components = X.shape[1]
+    lmnn = LargeMarginNearestNeighbor(init=init, n_components=n_components)
+    assert_raise_message(ValueError,
+                         'The preferred embedding dimensionality '
+                         '`n_components` ({}) does not match '
+                         'the output dimensionality of the given '
+                         'linear transformation `init` ({})!'
+                         .format(n_components, init.shape[0]),
+                         lmnn.fit, X, y)
+
+    # n_components > X.shape[1]
+    n_components = X.shape[1] + 2
+    lmnn = LargeMarginNearestNeighbor(init=init, n_components=n_components)
+    assert_raise_message(ValueError,
+                         'The preferred embedding dimensionality '
+                         '`n_components` ({}) cannot be greater '
+                         'than the given data dimensionality ({})!'
+                         .format(n_components, X.shape[1]),
+                         lmnn.fit, X, y)
+
+    # n_components < X.shape[1]
+    lmnn = LargeMarginNearestNeighbor(n_components=2, init='identity')
+    lmnn.fit(X, y)
+
+
+def test_init_transformation():
+    X, y = datasets.make_classification(n_samples=30, n_features=5,
+                                        n_redundant=0, random_state=0)
+    X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+    # Initialize with identity
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, init='identity')
+    lmnn.fit(X_train, y_train)
+
+    # Initialize with PCA
+    lmnn_pca = LargeMarginNearestNeighbor(n_neighbors=3, init='pca')
+    lmnn_pca.fit(X_train, y_train)
+
+    # Initialize with a transformation given by the user
+    init = np.random.rand(X.shape[1], X.shape[1])
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, init=init)
+    lmnn.fit(X_train, y_train)
+
+    # init.shape[1] must match X.shape[1]
+    init = np.random.rand(X.shape[1], X.shape[1] + 1)
+    lmnn = LargeMarginNearestNeighbor(n_neighbors=3, init=init)
+    assert_raise_message(ValueError,
+                         'The input dimensionality ({}) of the given '
+                         'linear transformation `init` must match the '
+                         'dimensionality of the given inputs `X` ({}).'
+                         .format(init.shape[1], X.shape[1]),
+                         lmnn.fit, X_train, y_train)
+
+    # init.shape[0] must be <= init.shape[1]
+    init = np.random.rand(X.shape[1] + 1, X.shape[1])
```

Comment:
same here (same comment for all `np.random.`)

Reply:
Done.